### DISCUSSION TITLE
How to profile all functions belong to a module?

### DISCUSSION THREAD

**Discussion by PaleNeutron** [pyutils/line_profiler#305] [Q&A]
I have a module called `foo` and a `test.py` file import functions from the module. Is there any way to profile all lines belong to `foo`?

I read https://kernprof.readthedocs.io/en/latest/line_profiler.autoprofile.html#auto-profiling , the description seems what I need  but if I use
 
```
python -m kernprof -lv -p foo main.py
```

It takes a long time and nothing happens.

**Comment by Erotemic** [comment]
@ta946

**Comment by ta946** [comment]
> ```
> python -m kernprof -lv -p foo main.py
> ```

that is the correct usage. although i personally use a different method
```
path_to_python_folder eg: C:/Program Files/Python38/Scripts
<path_to_python_folder>/kernprof.exe -lv -p foo main.py
```
i doubt that would make a difference but you can give it a try

Anyways, since the syntax is correct, that means its something from your side.

1. maybe your line_profiler is not installed correctly. to test that it works:
    1. create a file called `demo_primes2.py`
    2. enter the following code:
        ```python
        def is_prime(n):
            '''
            Check if the number "n" is prime, with n > 1.

            Returns a boolean, True if n is prime.
            '''
            max_val = n ** 0.5
            stop = int(max_val + 1)
            for i in range(2, stop):
                if n % i == 0:
                    return False
            return True


        def find_primes(size):
            primes = []
            for n in range(size):
                flag = is_prime(n)
                if flag:
                    primes.append(n)
            return primes


        def main():
            print('start calculating')
            primes = find_primes(100000)
            print(f'done calculating. Found {len(primes)} primes.')


        if __name__ == '__main__':
            main()
        ```
    3. run `python -m kernprof -lv -p demo_primes2.py demo_primes2.py`
    4. do you get line_profiler timing output?
    
2. it could be that `foo` or `main.py` has something weird happening in it that is interfering with autoprofiling. check that line_profiler without autoprofiling is working:
   1. add `@profile` before your `main()` function
        ```python
        @profile
        def main():
            ...
        ```
    2. run `python -m kernprof -lv main.py` (without `-p`)
    3. you can remove `@profile` now
    4. do you get line_profiler timing output?

if both of the above cases work and you are getting timing output, then there is something in `foo` or `main.py` interfering with autoprofiling.
we would need you to provide a minimum working code that can reproduce the issue.
if `foo` is a public module on pypi, can you tell us what it is?

I have encountered some weird instances where autoprofiling will act up when there is an external process that calls the script. eg: unittests.
Could potentially be an issue with non-daemon threads? i dont remember if that was a problem from my code and unrelated to line_profiler, just brainstorming..

we would need more information so we can help
